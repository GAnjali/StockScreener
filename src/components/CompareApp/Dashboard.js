import ValueForm from "./ValueForm";
import DatePicker from "./DatePicker";
import React from "react";

const Dashboard = (props) => {
    return (
        <div className="app">
            <header className="App-header">
                <h2 className="App-title">Compare Stocks for a Mutual fund</h2>
            </header>
            <div className="App-intro">
                <ValueForm className={"amount-value-form"} amountToInvest={props.amountToInvest}
                           label={"Amount to invest:"}
                           name={"amountToInvest"}
                           placeholder={"Enter the Amount"} handleChange={props.handleAmountInput}/>
                <DatePicker date={props.date}
                            handleDateChange={props.handleDateChange}/>
                <div id={"graph"}/>
            </div>
        </div>
    )
};

export default Dashboard;