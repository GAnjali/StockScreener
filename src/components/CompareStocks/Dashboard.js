import InputField from "./InputField";
import DatePicker from "./DatePicker";
import React from "react";

const Dashboard = (props) => {
    return (
        <div className="app">
            <header className="app-header">
                <h2 className="app-title">Compare Stocks for a Mutual fund</h2>
            </header>
            <div className="app-intro">
                <InputField className={"amount-value-form"} amountToInvest={props.amountToInvest}
                            label={"Amount to invest:"}
                            name={"amountToInvest"}
                            placeholder={"Enter the Amount"} handleChange={props.handleAmountInput}/>
                <DatePicker date={props.date}
                            handleDateChange={props.handleDateChange}/>
                <div id={"graph"}/>
            </div>
        </div>
    )
};

export default Dashboard;