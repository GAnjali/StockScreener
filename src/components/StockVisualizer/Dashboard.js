import StockSelector from "./StockSelector";
import DateRangeSelector from "./DateRangeSelector";
import React from "react";

const Dashboard = (props) => {
    return (
        <div className="app">
            <header className="app-header">
                <h2 className="app-title">Stock Visualizer</h2>
            </header>
            <div className="app-intro">
                <h3 className={"stock-title"}>Stock:</h3>
                <div className="stock-selection">
                    <StockSelector stocks={props.stocks} onSelectChange={props.handleSelectStock}/>
                    <br/><br/>
                </div>
                <h3 className={"date-range-title"}>Date Range:</h3>
                <div className="dateRange-selection">
                    <DateRangeSelector
                        startDate={props.selectedStartDate}
                        endDate={props.selectedEndDate}
                        handleSelectDate={props.handleSelectDate}/>
                </div>
            </div>
            <div id={"graph"}/>
        </div>
    )
};

export default Dashboard;