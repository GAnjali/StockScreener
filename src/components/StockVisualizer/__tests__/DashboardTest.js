import React from "react";
import {shallow} from "enzyme";
import Dashboard from "../Dashboard";

describe("Dashboard", () => {

    it("Should render without crashing", () => {
        shallow(<Dashboard />);
    });

    it('should match the Home snapshot', function () {
        const dashboardComponent = shallow(<Dashboard />);
        expect(dashboardComponent).toMatchSnapshot();
    });
});
