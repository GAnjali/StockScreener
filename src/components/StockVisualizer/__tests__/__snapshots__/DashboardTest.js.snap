// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockSelector should match the Home snapshot 1`] = `ShallowWrapper {}`;
